plugins/StereoControl/StereoControl.cpp

```diff
@@ -0,0 +1,443 @@
+				const float rGainL = panDualL < 0 ? 1.f + panDualL : 1.f;
+				const float lGainR = panDualR > 0 ? 1.f - panDualR : 1.f;
+				const float rGainR = panDualR < 0 ? 1.f + panDualR : 1.f;
+				const float temp = s[0];
+				s[0] = s[0] * lGainL + s[1] * lGainR;
+				s[1] = temp * rGainL + s[1] * rGainR;
+				break;
+			}
+			case 2:// Haas
+			{
+				float haasResult = 0;
+				float haasDelayVal[2] = {0};
+
+				if (pan >= 0)
+				{
+					haasDelayVal[0] = linearInterpolate(0, 0.0008f * m_sampleRate * panDelay, abs(pan));
+				}
+				else
+				{
+					haasDelayVal[1] = linearInterpolate(0, 0.0008f * m_sampleRate * panDelay, abs(pan));
+				}
+
+				for (int i = 0; i < 2; ++i)
+				{
+					if ((i == 0 && pan > 0) || (i == 1 && pan < 0))
+					{
+						// Read next value from delay buffer
+						float readLoc = m_haasIndex[i] - haasDelayVal[i];
+						if (readLoc < 0) {readLoc += m_haasBufSize;}
+						float readLocFrac = fraction(readLoc);
+						if (readLoc < m_haasBufSize - 1)
+						{
+							haasResult = linearInterpolate(m_haasBuf[i][floor(readLoc)], m_haasBuf[i][ceil(readLoc)], readLocFrac);
+						}
+						else// For when the interpolation wraps around to the beginning of the buffer
+						{
+							haasResult = linearInterpolate(m_haasBuf[i][m_haasBufSize - 1], m_haasBuf[i][0], readLocFrac);
+						}
+					}
+
+					// Increment delay ring buffer index
+					++m_haasIndex[i];
+					if (m_haasIndex[i] >= m_haasBufSize)
+					{
+						m_haasIndex[i] -= m_haasBufSize;
+					}
+					// Send new value to delay line
+					m_haasBuf[i][m_haasIndex[i]] = s[i];
+				}
+
+				if (pan != 0)
+				{
+					const float minGain = linearInterpolate(1.f, 0.15f, abs(pan));
+					if (pan >= 0)
+					{
+						const float lGain = ((1.f - pan) + minGain) / (1.f + minGain);
+						s[0] = haasResult * lGain;
+					}
+					else
+					{
+						const float rGain = ((1.f + pan) + minGain) / (1.f + minGain);
+						s[1] = haasResult * rGain;
+					}
+				}
+
+				const int lowChnl = (pan <= 0);
+				const int highChnl = !lowChnl;
+
+				float lp;
+				float hp;
+
+				// Filter parameters are semi-arbitrary, but are vaguely based on some random graphs I saw during research
+				float temp = -abs(pan) + 1.f;
+				float filtFreq = 2000.f + 8000.f * temp * temp;
+				float hpGain = abs(pan) < 0.5 ? 1.f - abs(pan) * 2.f : 0.f;
+				hpGain = linearInterpolate(1.f, hpGain, panSpectral);
+				multimodeFilter(s[lowChnl], multimodeCoeff(filtFreq), lp, hp, m_haasSpectralPanFilter[lowChnl]);
+
+				s[lowChnl] = lp + hp * hpGain;
+
+				filtFreq = 2000.f;
+				hpGain = abs(pan) + 1.f;
+				hpGain = linearInterpolate(1.f, hpGain, panSpectral);
+				multimodeFilter(s[highChnl], multimodeCoeff(filtFreq), lp, hp, m_haasSpectralPanFilter[highChnl]);
+
+				s[highChnl] = lp + hp * hpGain;
+			}
+		}
+
+		if (dc)
+		{
+			for (int i = 0; i < 2; ++i)
+			{
+				m_dcInfo[i] = m_dcInfo[i] * 0.999f + s[i] * 0.001f;
+				s[i] -= m_dcInfo[i];
+			}
+		}
```

Comment:
It seems like you're using a one-pole LP filter for DC removal. However, this won't make the output signal consistent for various sample rates.